kbase-extension/static/kbase/js/widgets/narrative_core/jobCommChannel.js

```diff
@@ -221,6 +254,113 @@ define([
+                    break;
+
+                // JOB messages
+                case 'job_canceled':
+                    this.sendBusMessage(JOB, msgData.job_id, 'job-canceled', {
+                        jobId: msgData.job_id,
+                        via: 'job_canceled',
+                    });
+                    break;
+
+                case 'job_comm_error':
+                    if (msgData) {
+                        jobId = msgData.job_id;
+                        switch (msgData.source) {
+                            case 'cancel_job':
+                                this.sendBusMessage(JOB, jobId, 'job-cancel-error', {
+                                    jobId,
+                                    message: msgData.message,
+                                });
+                                this.sendBusMessage(JOB, jobId, 'job-canceled', {
+                                    jobId,
+                                    error: msgData.message,
+                                });
+                                break;
+                            case 'job_logs':
+                            case 'job_logs_latest':
+                                this.sendBusMessage(JOB, jobId, 'job-log-deleted', {
+                                    jobId,
+                                    message: msgData.message,
+                                });
+                                this.sendBusMessage(JOB, jobId, 'job-logs', {
+                                    jobId,
+                                    error: msgData.message,
+                                });
+                                break;
+                            case 'job_retried':
+                                this.sendBusMessage(JOB, jobId, 'job-retried', {
+                                    jobId: jobId,
+                                    error: msgData.message,
+                                });
+                                break;
+                            default:
+                                this.sendBusMessage(JOB, jobId, 'job-error', {
+                                    jobId,
+                                    message: msgData.message,
+                                    request: msgData.source,
+                                });
+                                break;
+                        }
+                    }
+                    console.error('Error from job comm:', msg);
+                    break;
+
+                case 'job_does_not_exist':
+                    if (msgData.source === 'job_status') {
+                        this.sendBusMessage(JOB, msgData.job_id, 'job-status', {
+                            jobId: msgData.job_id,
+                            jobState: {
+                                job_id: msgData.job_id,
+                                status: 'does_not_exist',
+                            },
+                        });
+                        break;
```

Comment:
Instead of using `job-does-not-exist` for job status responses where the ID is nonexistent, use the `job-status` channel with the status `does_not_exist`.